addons/mail/models/reset_template_mixin.py

```diff
@@ -0,0 +1,106 @@
+                # (it will be missing e.g. when importing data-only modules using base_import_module)
+                path_info = get_resource_from_path(self._context['install_filename'])
+                if path_info:
+                    val['template_fs'] = '/'.join(path_info[0:2])
+        return super().create(vals_list)
+
+    def write(self, vals):
+        vals['is_template_modified'] = vals.get('is_template_modified', True)
+        return super().write(vals)
+
+    # ------------------------------------------------------------
+    # RESET TEMPLATE
+    # ------------------------------------------------------------
+
+    def _process_translation_data(self, trans_file, lang, xml_ids):
+        """Populates the ir_translation table.
+        :param trans_file: path to open a translation file
+        :param lang: language code of the translations contained in `trans_file`
+                     language must be present and activated in the database
+        """
+        with file_open(trans_file, mode='rb') as fileobj:
+            fileformat = os.path.splitext(trans_file)[-1][1:].lower()
+            reader = TranslationFileReader(fileobj, fileformat=fileformat)
+            # Process a single PO (or POT) entry
+            for row in reader:
+                if row.get('imd_name') in xml_ids:
+                    # copy Translation from Source to Destination object
+                    self.env['ir.translation']._set_ids(
+                        row['name'], 'model', lang, self._ids, row.get('value', ''), row['src'],
+                    )
+
+    def _override_translation_term(self, module_name, xml_ids):
+        for code, _ in self.env['res.lang'].get_installed():
+            lang_code = tools.get_iso_codes(code)
+            base_lang_code = lang_code.split('_')[0] if '_' in lang_code else lang_code
+            is_po_file_available = False
+            # In case of sub languages (e.g fr_BE), load the base language first, (e.g fr.po) and
+            # then load the main translation file (e.g fr_BE.po)
+
+            # Step 1: reset translation terms with base language file
+            base_trans_file = get_module_resource(module_name, 'i18n', base_lang_code + '.po')
+            if base_trans_file:
+                is_po_file_available = True
+                self._process_translation_data(base_trans_file, code, xml_ids)
+
+            # Step 2: reset translation file with main language file (can possibly override the
+            # terms coming from the base language)
+            trans_file = get_module_resource(module_name, 'i18n', lang_code + '.po')
+            if trans_file:
+                is_po_file_available = True
+                self._process_translation_data(trans_file, code, xml_ids)
+
+            # If no translation file available, unlink custom translated terms linked to template
+            if not is_po_file_available:
+                self.env['ir.translation'].search([
```

Comment:
Just to know: do people have the right to do so (search and unlink translation terms) ? I suppose yes as it can be done through interface, just asking to be sure.

Reply:
Employees have read rights so should be ok